### DISCUSSION TITLE
ER_UNKNOWN_ERROR: connection was bad

### DISCUSSION THREAD

**Discussion by NAZBIN** [directus/directus#12823] [Archive]
### Describe the Bug

![image](https://user-images.githubusercontent.com/37393662/163788473-0a2338b4-7684-4130-b013-87bf7d741caa.png)
![image](https://user-images.githubusercontent.com/37393662/163788509-bea6d437-472c-4924-a928-4643d29e4e3b.png)

### To Reproduce

This error occurs when creating tables and editing table structures.
I have tried the following pooling configurations, but none of them can solve this problem：
DB_POOL_MAX=7
DB_POOL_MIN=0
DB_POOL_idleTimeoutMillis=10000
DB_POOL_acquireTimeoutMillis=10000
DB_POOL_createRetryIntervalMillis=200
DB_IDLE=10000

### What version of Directus are you using?

9.6.0

### What version of Node.js are you using?

14.15.4

### What database are you using?

mysql5.7

### What browser are you using?

chrome

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
What is the environment variable value for `DB_CLIENT` and the other `DB_` config vars? It looks like you're trying to connect to a `mysql` database using the `pg` DB_CLIENT 🤔

**Reply by NAZBIN** [reply]
`
DB_CLIENT="mysql"
DB_HOST="xxxxxxx"
DB_PORT="xxxxxxx"
DB_DATABASE="directus"
DB_USER="xxxxxx"
DB_PASSWORD="xxxxxx"
DB_POOL_MAX=10
DB_POOL_MIN=0
DB_POOL_idleTimeoutMillis=10000
DB_POOL_acquireTimeoutMillis=10000
DB_POOL_createRetryIntervalMillis=200
DB_POOL_reapIntervalMillis=1000
DB_POOL_destroyTimeoutMillis=1000
DB_POOL_createTimeoutMillis=30000
DB_acquireConnectionTimeout=1000
`

I've tried all the variables set above, but I still haven't solved the error.  I can successfully connect to the database and perform the creation and editing of tables. However, errors will be reported when creating tables, editing table structures and deleting tables ('ER_UNKNOWN_ERROR: connection was bad').    need help😭

**Reply by rijkvanzanten** [reply]
How are you running MySQL? What happens if you remove all the DB_POOL configuration? (Also, the DB POOL configuration shouldn't use camelcase, for example `DB_POOL_createRetryIntervalMillis` should be `DB_POOL_CREATE_RETRY_INTERVAL_MILLIS` 👍🏻 )

**Reply by NAZBIN** [reply]
i use the production mysql,  The above error was reported before I added the DB_POOL attribute. I tried to pool configuration to solve this problem, but it hasn't been solved yet.   thanks bro (DB_POOL_CREATE_RETRY_INTERVAL_MILLIS) i try this way again ,Still failed to solve the bad connection problem 😭

**Reply by NAZBIN** [reply]
I found on the Internet that the connection pool needs to take the initiative to test whether the holding link is valid, but I don't know how to configure it in Directus

**Reply by NAZBIN** [reply]
”Affected by the active disconnection mechanism of online database. As long as the connection pool regularly detects whether the link is valid, this problem can be avoided“  but I don't know how to configure it in Directus 😓

**Reply by rijkvanzanten** [reply]
> i use the production mysql

What "production" MySQL? What versions? Where is it hosted? How are you keeping it alive? How is the network configured? There's loads of network infrastructure configurations that might affect this, so it's impossible for me to tell what configuration is needed to make things work in your specific environment without all the details 😄 For what it's worth, I've never seen this particular error message before, so it feels like there must be something unusual going on in this specific setup 🤔

**Reply by NAZBIN** [reply]
> > mysql version 5.7
At first, I was connected to the local database. After everything was normal, I applied for an online database connection in the company, and other configurations have not been changed. Then I will report an error when I operate the table. Yes, I looked through all the problems in directus and didn't find this error . so headache...

**Reply by rijkvanzanten** [reply]
Does this company database have any firewalls or other services in between that might mess with the connection? In any case, I'm afraid I won't be able to really help with this, as it sounds like there's a very specific corporate setup you're working with that might need some special configuration magic that i'm not aware of :C

**Reply by NAZBIN** [reply]
The problem is with the connection， so Is there any way to automatically test the connection of the database every 30 seconds in directus, and delete and rebuild if it is not available ?
